syntax = "proto3";

import "google/protobuf/any.proto";

package torch2trt;


enum DataTypeMsg {
  kFloat = 0;
  kHalf = 1;
  kInt8 = 2;
  kInt32 = 3;
}

message TensorShapeMsg {
  repeated int64 size = 1; // does not include batch
}

// BEGIN TORCH METHODS


// END TORCH METHODS

message TorchPluginMsg {
  DataTypeMsg dtype = 1;
  repeated TensorShapeMsg input_shapes = 2;
  repeated TensorShapeMsg output_shapes = 3;
  
  google.protobuf.Any method = 4;
}